Sources/SwiftBSON/BSONDocument.swift

```diff
@@ -16,21 +16,6 @@ public struct BSONDocument {
 
     private var storage: BSONDocumentStorage
 
-    internal var byteLength: Int {
```

Comment:
this seemed better suited to living on the `BSONDocumentStorage` type rather than on `BSONDocument` so I've moved it there.
I also renamed it `encodedLength` as I was having a hard time keeping track of what `byteLength` meant and I think this better conveys its meaning